### PR TITLE
Make Invoke label case insensitive and include filename in error message.

### LINKED ISSUE #96: named invokes should have a case insensitive match
As reported by Chris Maynard ... Upper case and lower case are treated as being different when looking for matching strings. The check should be case independent to fit in with the fact that Fortran is case independent.

```
module alg
  use field_mod, only: field_type
  implicit none
  subroutine mini_alg
    implicit none
    type(field_type) :: f1, f2, f3
    call invoke( name = "X_times_Y", &
                 X_times_Y(f1, f2, f3)  &
                  )
    call invoke( name = "x_times_y", &
                 X_times_Y(f2, f3, f1)  &
                  )
  end subroutine mini_alg
end module alg
```

If he runs PSyclone over this, it generates the same routine twice with the same name

### PR DESCRIPTION
**Pull request by arporter**
Fixes #96 and #94.

### CODE REVIEW

File: doc/getting_going.rst [hunk 1]
```diff
@@ -55,6 +55,7 @@ Note that some Linux distributions install pip only for python 3 by
 default. In this case it is necessary to install pip for python 2. For
 example in openSUSE 42.2:
 ::
+
```
Comment (rupertford) [inline doc/getting_going.rst]: unrelated improvement. Good.

File: src/psyclone/tests/parse_test.py [hunk 2]
```diff
@@ -37,6 +37,11 @@
 ''' A module to perform pytest unit and functional tests on the parse
 function. '''
 
+# Since this is a file containing tests which often have to get in and
+# change the internal state of objects we disable pylint's warning
+# about such accesses
+# pylint: disable=protected-access
```
Comment (rupertford) [inline src/psyclone/tests/parse_test.py]: Nice.

File: src/psyclone/tests/parse_test.py [hunk 3]
```diff
@@ -208,17 +215,36 @@ def test_duplicate_named_invoke():
...
+    assert "3.4_multi_invoke_name_clash_case_insensitive.f90" in str(err)
 
 
 def test_get_stencil():
     ''' Check that parse.get_stencil() raises the correct errors when
     passed various incorrect inputs '''
     from psyclone.parse import get_stencil
     from psyclone.expression import ExpressionNode, FunctionVar
-    node = ExpressionNode(["1"])
+    enode = ExpressionNode(["1"])
```
Comment (rupertford) [inline src/psyclone/tests/parse_test.py]: indicate this is different from a generic node by changing the name. OK.

File: doc/algorithm_layer.rst [hunk 4]
```diff
@@ -106,7 +106,7 @@ call like so:
 
   ...
   call invoke(integrate_one_kernel(arg1,arg2), &
-              name="compute something")
```
[the remark this answers is not in the document]
Comment (arporter) [reply doc/algorithm_layer.rst]: OK.

File: doc/algorithm_layer.rst [hunk 5]
```diff
@@ -115,11 +115,13 @@ replaced by '_' characters) is used to name the corresponding
...
 the generated PSy-layer subroutine will be named
 "invoke_compute_something." Each invoke label must currently be unique
-within an Algorithm source file. In the future it is intended that
-this name will allow invoke-specific optimisations to be applied as
-well as enabling more readable profiling output. It may also be used
-to instruct PSyclone to just generate a single subroutine to implement all
-invokes that share the same label.
+within an Algorithm source file. Note that, in keeping with the
+Fortran language, labels are not case sensitive. In the future it is
+intended that the labelling of invokes will allow invoke-specific
```
[the remark this answers is not in the document]
Comment (arporter) [reply doc/algorithm_layer.rst]: Done.

File: doc/dynamo0p3.rst [hunk 6]
```diff
@@ -44,7 +44,7 @@ objects and their use are discussed in the following sections.
                builtin1(scalar1, field2, field3),                &
                kernel2(field1, stencil_extent, field3, scalar1), &
 	       assembly_kernel(cma_op1, operator1),              &
-               name="some calculation"                           &
+               name="some_calculation"                           &
```
[the remark this answers is not in the document]
Comment (arporter) [reply doc/dynamo0p3.rst]: OK.

File: src/psyclone/tests/test_files/dynamo0p3/3.4_multi_invoke_name_clash_case_insensitive.f90 [hunk 7]
```diff
@@ -0,0 +1,59 @@
+!-------------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (arporter) [reply src/psyclone/tests/test_files/dynamo0p3/3.4_multi_invoke_name_clash_case_insensitive.f90]: I've cut this down.

**Comment by arporter** [comment]
Note that we prepend "invoke_" to any label in order to generate the name of the PSy-layer routine. This reduces but does not eliminate the restrictions we must apply. e.g. there's currently nothing to stop a user putting brackets of any description in the string.

**Comment by arporter** [comment]
Ready for second review by @rupertford

**Pull request merged**